apps/avifdec.c

```diff
@@ -378,11 +386,12 @@ int main(int argc, char * argv[])
     returnCode = 0;
 
 cleanup:
-    if (returnCode != 0) {
+    if (returnCode != 0 && decoder != NULL) {
```

Comment:
Just wanted to confirm that you prefer to not nest these two if statements as follows:

```
    if (decoder != NULL) {
        if (returnCode != 0) {
            avifDumpDiagnostics(&decoder->diag);
        }
        avifDecoderDestroy(decoder);
    }
```

Reply:
Done